libraries/heft-config-file/src/ConfigurationFile.ts

```diff
@@ -91,17 +91,41 @@ interface IConfigurationFileFieldAnnotation<TField> {
   originalValues: { [propertyName in keyof TField]: unknown };
 }
 
+/**
+ * Options provided to the custom resolver specified in {@link ICustomJsonPathMetadata}.
+ *
+ * @beta
+ */
+export interface IJsonPathMetadataResolverOptions<TConfigurationFile> {
+  /**
+   * The name of the property being resolved.
+   */
+  propertyName: string;
+  /**
+   * The value of the path property being resolved.
+   */
+  propertyValue: string;
+  /**
+   * The path to the configuration file the property was obtained from.
+   */
+  configurationFilePath: string;
+  /**
+   * The configuration file the property was obtained from.
+   */
+  configurationFile: Partial<TConfigurationFile>;
```

Comment:
Partial?

Reply:
Yeah. Technically, we may not have a full `TConfigurationFile` at this point, since the base configuration may not be valid in the schema. This is also why we only validate the schema after the entire configuration file stack is loaded.